TCAT/Cells/ServiceAlertTableViewCell.swift

```diff
@@ -0,0 +1,238 @@
+        if let fromDate = alert?.fromDate, let toDate = alert?.toDate {
+            timeSpanLabel.text = formatTimeString(fromDate, toDate: toDate)
+        }
+        descriptionLabel.text = alert?.message
+        if let routes = alert?.routes, !routes.isEmpty {
+            setupAffectedRoutesStackView()
+            setupaffectedRoutesLabel()
+        }
+        
+        if rowNum > 0 {
+            setupTopSeparator()
+        }
+    }
+    
+    private func setupTimeSpanLabel() {
+        
+        timeSpanLabel = UILabel()
+        timeSpanLabel.numberOfLines = 0
+        timeSpanLabel.font = .getFont(.semibold, size: 18)
+        timeSpanLabel.textColor = Colors.primaryText
+        
+        contentView.addSubview(timeSpanLabel)
+    }
+    
+    private func setupDescriptionLabel() {
+        
+        descriptionLabel = UILabel()
+        descriptionLabel.numberOfLines = 0
+        descriptionLabel.font = .getFont(.regular, size: 14)
+        descriptionLabel.textColor = Colors.primaryText
+        
+        
+        contentView.addSubview(descriptionLabel)
+    }
+    
+    private func setupaffectedRoutesLabel() {
+        
+        affectedRoutesLabel = UILabel()
+        affectedRoutesLabel.font = .getFont(.semibold, size: 18)
+        affectedRoutesLabel.textColor = Colors.primaryText
+        affectedRoutesLabel.text = "Affected Routes"
+        
+        contentView.addSubview(affectedRoutesLabel)
+    }
+    
+    private func setupAffectedRoutesStackView() {
+        
+        if var routes = alert?.routes, !routes.isEmpty {
+            affectedRoutesStackView = UIStackView()
+            for _ in 0..<rowCount() {
+                var subviews = [BusIcon]()
+                for _ in 0..<maxIconsPerRow() {
+                    if !routes.isEmpty {
+                        subviews.append(BusIcon(type: .directionSmall, number: routes.removeFirst()))
+                    }
+                }
+                let rowStackView = UIStackView(arrangedSubviews: subviews)
+                rowStackView.axis = .horizontal
+                rowStackView.spacing = 10
+                rowStackView.alignment = .leading
+                affectedRoutesStackView?.addArrangedSubview(rowStackView)
+            }
+        }
+        guard let stackView = affectedRoutesStackView else { return }
+        stackView.axis = .vertical
+        stackView.alignment = .top
+        stackView.spacing = 10
+        contentView.addSubview(stackView)
+    }
+    
+    private func setupTopSeparator() {
+        
+        topSeparator = UIView()
+        topSeparator?.backgroundColor = Colors.backgroundWash
+        
+        contentView.addSubview(topSeparator!)
+    }
+    
+    override func updateConstraints() {
+        
+        if let topSeparator = topSeparator {
+            topSeparator.snp.remakeConstraints { (make) in
+                make.top.leading.trailing.equalToSuperview().labeled("topSeparator: Top, Leading, Trailing")
```

Comment:
I've never seen this syntax! What does it do / what are the benefits over the usual way we use SnapKit?

Reply:
Helps a lot with debugging. Whenever there are constraint errors or warnings, instead of giving a constraint identifier which is a weird number attached to a view, it prints the label you pass in instead. I'll take them out tho, they def make the code look more cluttered.